### PR TITLE
nginx: add handler to restart nginx when configuration changes

### PR DESCRIPTION
**Pull request by TomasHofman**
@rpelisse @spyrkob am I correct to think that this should cause nginx to be restarted when some tasks cause modifications?

### CODE REVIEW

**Comment by spyrkob** [comment]
I think you need add 
```
notify:
    - Restart nginx
```
on tasks that require restart

**Comment by rpelisse** [comment]
@TomasHofman Yes, that is how it's working. It funny, I really thought I came around adding this handler...

**Comment by TomasHofman** [comment]
OK, I added some notifies here and there, but I'm not sure if it's working. `ansible-playbook --syntax-check zeus.yml` doesn't seem to check the nested roles, or at least I'm not able to break it. I will need to investigate more on testability later....

**Review by spyrkob — APPROVED** [review]

File: roles/nginx/tasks/http_cache_setup.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@
   loop:
     - { src: templates/commons.d/maven.cache.conf.j2, dest: "{{ nginx_config_dir }}/commons.d/maven.cache.conf" }
     - { src: templates/commons.d/caches.conf.j2, dest: "{{ nginx_config_dir }}/commons.d/caches.conf" }
+  notifies:
```
Comment (spyrkob) [inline roles/nginx/tasks/http_cache_setup.yml:18]: @TomasHofman that should be `notify`, sorry missed it earlier

File: roles/nginx/tasks/nginx_config.yml [hunk 2]
```diff
@@ -57,7 +57,9 @@
         - { src: templates/nginx.conf.j2, dest: "{{ nginx_config_dir }}/nginx.conf" }
         - { src: templates/commons.d/sites.conf.j2, dest: "{{ nginx_config_dir }}/commons.d/sites.conf" }
         - { src: templates/commons.d/proxy.conf.j2, dest: "{{ nginx_config_dir }}/commons.d/proxy.conf" }
-  when: nginx_config_dir_path is defined 
+      notifies:
```
Comment (spyrkob) [inline roles/nginx/tasks/nginx_config.yml:60]: same as above

File: roles/nginx/tasks/ssl.yml [hunk 3]
```diff
@@ -73,3 +73,5 @@
     dest: "{{ nginx.home }}/config/commons.d/ssl.conf"
     owner: nginx
     group: nginx
+  notifies:
```
Comment (spyrkob) [inline roles/nginx/tasks/ssl.yml:76]: same as above

**Review by spyrkob — CHANGES_REQUESTED** [review]

File: roles/nginx/tasks/http_cache_setup.yml (hunk 1 above)
Comment (TomasHofman) [reply roles/nginx/tasks/http_cache_setup.yml]: Sorry, PR updated :)

**Comment by spyrkob** [comment]
thanks :)

**Pull request merged**